src/main/java/org/apache/commons/lang3/builder/DiffBuilder.java

```diff
@@ -842,6 +842,10 @@ public DiffBuilder append(final String fieldName, final Object lhs,
         }
 
         // Not array type
+        if (left != null ? left.equals(right) : right.equals(left)) {
```

Comment:
```left != null``` will always be true, since the constructor prevents the field from getting initialized with null. If that wouldn't be the case, the proposed code could still throw NPE if left and right are both null.